workflow/rules/diffexp.smk

```diff
@@ -264,15 +264,19 @@ rule plot_diffexp_heatmap:
 rule plot_heatmap_predefined:
     input:
         logcountmatrix_file="results/tables/logcount-matrix/{model}.logcount-matrix.tsv",
-        predef_genelist="resources/selected_gene_from_ref.txt",
+        predef_genelist=config["diffexp"]["pre-define-genelist"],
     output:
-        predefgene_png="results/heatmaps/predefgenes_heatmap.png",
+        predefgene_pdf=report(
```

Comment:
Noted. updated the code.

Reply:
But the output file is now still named differently here (predefgene_pdf). If you name it the same in both rules, the script can be even simpler, because the writing is the same. just the input to the heatmap function is generated differently.

Reply:
Now we have single rule and same output file for both context. Depending on the `predef_genelist` the script will produce the heatmap.